compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -2377,6 +2375,9 @@ private STNode parseTypeDescriptorInternal(List<STNode> qualifiers, ParserRuleCo
                 reportInvalidQualifierList(qualifiers);
                 STNode distinctKeyword = consume();
                 return parseDistinctTypeDesc(distinctKeyword, context);
+            case TRANSACTION_KEYWORD:
```

Comment:
Do we reach here? if there's a `transaction:` isQualifiedIdentifierPredeclaredPrefix() call above should catch neh?